src/internal/utils/parseConfigsFromDocument.ts

```diff
@@ -0,0 +1,44 @@
+import { ThemeData } from "../types/themeData.ts";
+import { Data } from "@measured/puck";
+import { resolveVisualEditorData } from "../../utils/resolveVisualEditorData.ts";
+import { jsonFromEscapedJsonString } from "./jsonFromEscapedJsonString.ts";
+
+type VisualEditorData = {
+  visualConfigurationData?: Data;
+  themeData?: ThemeData;
+};
+
+export const parseConfigsFromDocument = (
+  document: any,
+  pageSet: string,
+  siteId: number
+): VisualEditorData => {
+  if (document) {
+    const resolvedData = resolveVisualEditorData(
+      { document: document },
+      pageSet,
+      siteId
+    );
+    const visualConfigurationData = resolvedData.document.visualTemplate;
+    const themes: Record<string, any>[] =
+      resolvedData.document._site.pagesTheme;
+
+    let themeDataString: string = "{}";
```

Comment:
This is interesting? idk if it'd be cleaner to do return 
      visualConfigurationData: visualConfigurationData,
      themeData:{},
 if the themes.length = 0.
 
but also can ignore this, overall lgtm